src/Rings/localization_interface.jl

```diff
@@ -0,0 +1,361 @@
+############################################################################
+
+@Markdown.doc """
+    AbsLocalizedIdeal{RingType, RingElemType, MultSetType} <: Ideal{RingElemType}
+
+Abstract type for finitely generated ideals ``IS⁻¹ ⊂ R[S⁻¹]`` in localized rings. 
+"""
+abstract type AbsLocalizedIdeal{RingType, RingElemType, MultSetType} <: Ideal{RingElemType} end
+
+### required getter functions
+function gens(I::AbsLocalizedIdeal{RingType, RingElemType, MultSetType}) where {RingType, RingElemType, MultSetType}
+  error("`gens(I)` has not been implemented for `I` of type $(typeof(I))")
+end
+
+function base_ring(I::AbsLocalizedIdeal{RingType, RingElemType, MultSetType}) where {RingType, RingElemType, MultSetType}
+  error("`base_ring(I)` has not been implemented for `I` of type $(typeof(I))")
+end
+
+### required constructors
+function ideal(
+    W::AbsLocalizedRing{RingType, RingElemType, MultSetType}, 
+    f::AbsLocalizedRingElem{RingType, RingElemType, MultSetType} 
+  ) where {RingType, RingElemType, MultSetType}
+  error("`ideal(W, f)` has not been implemented for `W` of type $(typeof(W)) and `f` of type $(typeof(f))")
+end
+
+function ideal(
+    W::AbsLocalizedRing{RingType, RingElemType, MultSetType}, 
+    v::Vector{AbsLocalizedRingElem{RingType, RingElemType, MultSetType}}
+  ) where {RingType, RingElemType, MultSetType}
+  error("`ideal(W, v)` has not been implemented for `W` of type $(typeof(W)) and `v` of type $(typeof(v))")
+end
+
+function ideal(
+    W::AbsLocalizedRing{RingType, RingElemType, MultSetType}, 
+    f::RingElemType 
+  ) where {RingType, RingElemType, MultSetType}
+  error("`ideal(W, f)` has not been implemented for `W` of type $(typeof(W)) and `f` of type $(typeof(f))")
+end
+
+function ideal(
+    W::AbsLocalizedRing{RingType, RingElemType, MultSetType}, 
+    v::Vector{RingElemType}
+  ) where {RingType, RingElemType, MultSetType}
+  error("`ideal(W, v)` has not been implemented for `W` of type $(typeof(W)) and `v` of type $(typeof(v))")
+end
+
+### required functionality
+function Base.in(
+    f::AbsLocalizedRingElem{RingType, RingElemType, MultSetType}, 
+    I::AbsLocalizedIdeal{RingType, RingElemType, MultSetType}
+  ) where {RingType, RingElemType, MultSetType}
+  error("`in(f, I)` has not been implemented for `f` of type $(typeof(f)) and `I` of type $(typeof(I))")
+end
+
+function Base.in(
+    f::RingElemType, 
+    I::AbsLocalizedIdeal{RingType, RingElemType, MultSetType}
+  ) where {RingType, RingElemType, MultSetType}
+  error("`in(f, I)` has not been implemented for `f` of type $(typeof(f)) and `I` of type $(typeof(I))")
+end
+
+
+### A catchall implementation for the ideal arithmetic 
+function Base.:*(I::T, J::T) where {T<:AbsLocalizedIdeal}
+  W = base_ring(I) 
+  W == base_ring(J) || error("the given ideals do not belong to the same ring")
+  new_gens = Vector{elem_type(W)}()
+  for f in gens(I) 
+    for g in gens(J)
+      push!(new_gens, f*g)
+    end
+  end
```

Comment:
You could also write this (matter of taste)
```suggestion
  for f in gens(I), g in gens(J)
    push!(new_gens, f*g)
  end
```

Reply:
`new_gens = [f*g for f in gens(J) for g in gens(I)]`

Reply:
Thanks! I was wondering how to implement a double for-loop in [...].